app/types/types.ts

```diff
@@ -0,0 +1,99 @@
+import { Model, Document } from "mongoose";
+import { STVEvent } from '../algorithms/stv';
+
+export enum ElectionSystems {
+    NORMAL = 'normal',
+    STV = 'stv',
+}
+
+export enum Status {
+    resolved = 'RESOLVED',
+    unresolved = 'UNRESOLVED',
+}
+
+export type Count = { [key: string]: number };
+
+export type ElectionResult = {
+    result: STVResult | NormalResult;
+    thr: number;
+    seats: number;
+    voteCount: number;
+    blankVoteCount: number;
+    useStrict: boolean;
+    log: STVEvent[] | Count;
+};
+
+export interface STVResult {
+    status: Status;
+    winners: AlternativeType[];
+}
+
+export interface NormalResult extends STVResult {
+    status: Status;
+    winners: AlternativeType[];
+    count: number;
+}
+
+interface IAlternative {
+    _id: string;
+    description: string;
+    election: ElectionType;
+}
+export type AlternativeType = IAlternative & Document;
```

Comment:
Why doesn't this extend `Document`?

Reply:
Because we want to have both the type with and without `Document`, the `AlternativeType` is used for the model when we need access to methods such as .find or .save etc., while `IAlternative` is to be used otherwise